### ISSUE TITLE
"Peers" console poorly formatted

### ISSUE THREAD

**Issue by lunokhod** [Anoncoin/oldrepo-backup#64]
As shown in these two images, the "peers" console is poorly formatted.

https://img.bi/#/YgB54BV!kNWFA0QBYTpRZ68duG8YdqgRWzhoHKux5P3skDrP
https://img.bi/#/B6nMs3o!88Z1UMIemNcEoLIa7Y0VxF2LuNHgXBdeu1X4tGlg

Not sure what we can do about this.

**Comment by GroundRod** [comment]
This is what I needed to see, thank you.  Clearly a problem on windows and linux builds too.  Wondered about it for the mac.  

& good topic for a new issue...as this may take awhile to get right.

The Peertable could be removed for a production build, until it is more ready for the public, it is a v9.99 level technology service, and I've only partially gotten the code changes done, so it will work at all with our modified v9.3 engine, from which we are based.

Additional data structures, class definitions & some coding work needs to be completed in several source code files, so as to be functional for all the data fields and meet the requirements adding it, forced me to introduce.

On the peer information display, Sync height may be the only one left that is not yet working properly, although Ban score may also need to be fixed.  Other than that, the information is pretty useful at a glance.

Obviously looks horrible when attempting to display full I2P destination addresses though.

I use it now, as is, for a network issues debugging tool, and soon will have b32.i2p addresses shown there, that will help some.  Instead of the full i2p destinations, we see now.  At that point, what we do next is up for debate.

Programming changes needed so the Peertable QT form functionality works better for the user, I'm delaying, if anyone has a simple quick fix, please us know.  From what I've looked at, it could easily turn into several days of work.  A more adjustable form layout for the user would be nice, such as an adjustable slider between the left and right display areas, the additional time to develop it, will not likely be a simple fix for me though, & such settings would need to be saved and restored the next time the program is run.  

Going to another control widget, one that lives inside of the peertable view, for where ever an address needs to be displayed, could replace the text now being shown, with a small scrollable window, one that would allow the user to see the 1st part of an address at least.  Changes like that would have be thoroughly tested & take sometime to implement.

**Comment by GroundRod** [comment]
...on your posted view, showing a peers detail:

This is the first time I've seen the unknown network services display problem fixed, had changed the code, and keep forgetting to look.  Is everyone happy with my decision to call network service bit #2  BLOOM.  Is that clear?  How about I2PNET for bit 7?  Or do you want to call them something else?

There was also a change I made, from calling service bit #1 NETWORK , as it was referred to.  Now for us, I'm using the term CLEARNET.  That seemed like a more descriptive word to me, as to what that bit really means for software such as ours, which supports multiple alternative networking protocols.  

It should be possible now to conduct some tests, running two v9 nodes, one configured for i2ponly, it should appear to the other node as only having the two services, BLOOM and I2PNET.  Same goes for a setup where only CLEARNET connections are possible, from that node, a peer should not see the I2PNET service bit enabled.

Any node broadcasting services it can not support, should have it's software fixed, and stop wasting valuable network time.  It is my intention to code it so such nodes get banned and dropped.

Also this should allow the user to identify those nodes which are running mixed, In that case, if the peers are initially connected via CLEARNET, I intend on developing a way for them to switch to I2P, and be more secure.

**Comment by lunokhod** [comment]
The word "Bloom" is not obvious to me, but maybe it is to an expert.

**Comment by Cryptoslave** [comment]
Bloom I dont get it neither.
Yeah B32 address would be best, cannot you display them in a box with slider?

**Comment by GroundRod** [comment]
Yes we may have to  move the addresses into slider boxes, even after they are reduced to b32 format.  Problem is they are now dynamically printed as text all over the forms face, as needed.  

The list on the left may look good, without needing to have a scroll bar for each and every address, it would cut in half (more) the number of addrs shown, if we have to use scrollbars there.  Plus its a virtual area which scrolls itself, when enough peers show up.

That leaves two fields to setup near the top of the peer details on the right for addresses.  Normally it is blank if no peer is selected, then it becomes what is shown to you, when clicking on one of entries in the left area, to select that peers details.  There are two addresses there now, however in the future I hope to have 3 addresses, if the peer is running mixed mode, I want to discover it's I2P address, so I can switch from a clearnet connection to an I2P connection with it.

So either a new form template has to be designed, with hard fixed locations for all this stuff, particularly the peer details, or one control structure is built in software and then moved around as needed while your navigating between the peers.  All the while, still keeping a good look for clearnet peers.  Its a good example of one challenging task, which needs to be done, if we are to keep the peertable view in the software.

**Comment by Cryptoslave** [comment]
yeah we shall keep the peertable.
Well dont worry about slider, at first just show b32 address it may be ok already (I was talking about the right screen btw, not the left one)

Are you sure it is good to link an i2p address to a clearnet IP? iirc lunokhod was against this, it may be not good in fact, is it possible for this to be avoided (or is it easy and a vector of attack to deanomize all mixed nodes?)

**Comment by GroundRod** [comment]
Very good point, had not thought of that.  Yes we need do nothing in this regard, it was just an idea I wanted to float.  ATM there is no way for us to know what they have for an I2P address in mixed mode, if we are connecting via clearnet.  All we can tell is that they are capable of that protocol service.
